File: Algorithm-Sources/Codeforces/Codeforces Round #225 Div.2/B. Multitasking.cpp
#include<stdio.h>
#include<iostream>
#include<set>
using namespace std;

int arr[1000][100];
int ans[10000][2];
int nAns = 0;

int main() {
//    freopen("input.in", "r", stdin);
    int n, m, k;
    scanf("%d%d%d", &n, &m, &k);
    
    for(int i = 0;i<n;i++) {
        for(int j = 0;j<m;j++)
            scanf("%d", &arr[i][j]);
    }
    
    for(int i = 0;i<m-1;i++) {
        bool check = false;
        set<pair<int, int> > tempAns;
        for(int l = 0;l<n;l++) {
            int v = arr[l][i];
            int idx = i;
            for(int j = i+1;j<m;j++) {
                if(k == 0 && v > arr[l][j]) {
                    v = arr[l][j];
                    idx = j;
                }
                else if(k == 1 && v < arr[l][j]) {
                    v = arr[l][j];
                    idx = j;
                }
            }
            if(idx != i) {
                check = true;
                if(k == 0)
                    tempAns.insert(make_pair(i, idx));
                else
                    tempAns.insert(make_pair(idx, i));
            }
        }
        for(set<pair<int, int> >::iterator j = tempAns.begin();j != tempAns.end(); j++) {
            int a = j->first, b = j->second;
            ans[nAns][0] = a;
            ans[nAns][1] = b;
            nAns++;
            for(int l = 0;l<n;l++) {
                if(arr[l][a] > arr[l][b])
                    swap(arr[l][a], arr[l][b]);
            }
        }
    }
    
    printf("%d\n", nAns);
    for(int i = 0;i<nAns;i++)
        printf("%d %d\n", ans[i][0]+1, ans[i][1]+1);
}

File: Algorithm-Sources/Codeforces/Codeforces Round #225 Div.2/B. Multitasking.py
n, m, k = [int(x) for x in raw_input().split()]

arr = []
for _ in xrange(n):
    ar = [int(x) for x in raw_input().split()]
    arr.append(ar)

ans = []

for i in xrange(m-1):
    check = False
    tempAns = set()
    for l in xrange(n):
        v = arr[l][i]
        idx = i
        for j in xrange(i+1, m):
            if k == 0 and v > arr[l][j]:
                v = arr[l][j]
                idx = j
            elif k == 1 and v < arr[l][j]:
                v = arr[l][j]
                idx = j
        if idx != i:
            check = True
            if k == 0:
                tempAns.add((i, idx))
            else:
                tempAns.add((idx, i))

    for a, b in tempAns:
        ans.append((a+1, b+1))
        for l in xrange(n):
            if arr[l][a] > arr[l][b]:
                arr[l][a], arr[l][b] = arr[l][b], arr[l][a]


print len(ans)
for a, b in ans:
    print a, b